src/remotes.c

```diff
@@ -172,9 +172,14 @@ remote_smb(char *address, char *options)
 	char *roptions = (char *)NULL;
 
 	if (ruser) {
-		roptions = (char *)xnmalloc(strlen(ruser) + strlen(options) + 11,
-															sizeof(char));
-		sprintf(roptions, "username=%s,%s", ruser, options);
+		if (options) {
+			roptions = (char *)xnmalloc(strlen(ruser) + strlen(options) + 11, sizeof(char));
+			sprintf(roptions, "username=%s,%s", ruser, options);
+		} else {
+			roptions = (char *)xnmalloc(strlen(ruser) + 11, sizeof(char));
```

Comment:
Your C-fu is fine. Just one little correction: in the false branch of the if statement 11 should be 10, because we don't have the comma (of the true branch) anymore, so 1 char less.